### PR TITLE
Update devtools window prop name

### PR DESCRIPTION
**Pull request by wtho**
* update prop `devToolsExtension` to `__REDUX_DEVTOOLS_EXTENSION__`
* add package redux-devtools-extension for typings
* define some typings for window

The listen-callback still works (at least `START` was fired on load, not sure how to test `STOP`), and there were no issues with the extension. Dispatching from the extension worked flawlessly.

I added the package `redux-devtools-extension` to give users of this library intellisense for the option types when initializing the devtools-enhancer. I decided against importing `devtoolsEnhancer` from the same package, as it kind of hides if the devtools were connected successfully or not, and the `isEnabled`-function in `@angular-redux/store` would still be required and does a very similar check. So better to rely only on `isEnabled` if we require it anyways.

### CODE REVIEW

File: packages/store/src/components/dev-tools.ts [hunk 1]
```diff
@@ -41,11 +54,11 @@ export class DevToolsExtension {
       }
     });
 
-    return environment.devToolsExtension(options);
+    return environment.__REDUX_DEVTOOLS_EXTENSION__!(options || {});
```
[the remark this answers is not in the document]
Comment (wtho) [reply packages/store/src/components/dev-tools.ts]: The type definition of `redux-deftools-extension` does not allow `undefined`, so I wanted to provide an empty object (`Argument of type 'EnhancerOptions | undefined' is not assignable to parameter of type 'EnhancerOptions'.`).

Checking the [devtools source](https://github.com/zalmoxisus/redux-devtools-extension/blob/48ac2d293366fdfb56378c6c36eaca1e65940fcd/src/browser/extension/inject/pageScript.js#L29-L33) there seems to be no issue with the config being undefined (called `reducer` there, which can also be undefined), so we can augment their type definitions to allow undefined. I think the easier approach is to keep it `options || {}` though (or defaulting to an emptpy object in the signature: `enhancer = (options: EnhancerOptions = {})`).

File: packages/store/src/components/dev-tools.ts [hunk 2]
```diff
@@ -1,9 +1,22 @@
+// tslint:disable:no-implicit-dependencies
```
[the remark this answers is not in the document]
Comment (wtho) [reply packages/store/src/components/dev-tools.ts]: Agree! I do not know why tslint triggered an error for me before. Will fix this.

File: packages/store/src/components/dev-tools.ts [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
 import { ApplicationRef, Injectable, NgZone } from '@angular/core';
-import { Unsubscribe } from 'redux';
+import { AnyAction, StoreEnhancer, Unsubscribe } from 'redux';
+import { EnhancerOptions } from 'redux-devtools-extension';
 import { NgRedux } from './ng-redux';
 
-declare const window: any;
-const environment: any = typeof window !== 'undefined' ? window : {};
+interface WindowWithReduxDevTools extends Window {
```
[the remark this answers is not in the document]
Comment (wtho) [reply packages/store/src/components/dev-tools.ts]: True, somehow I did not think of backwards-support. Will add it!

**Comment by wtho** [comment]
Updated backwards compatibility and removed lint-ignore.

Btw, this closes #56.

**Comment by fvilers** [comment]
Hi! Great work guys. Any estimation on a release date?

**Pull request merged**